From 94d155c7b6e0e66778b157794e4237fa7c69c066 Mon Sep 17 00:00:00 2001
From: duelu <luyizhou1@huawei.com>
Date: Tue, 3 Feb 2026 11:49:22 +0800
Subject: [PATCH] urma: bugfix of uvs and urma_admin 1. add parameter check of
 uvs_get_device_name_by_eid 2. fix the description of uvs_get_route_list 3.
 fix the return value of exec_cmd

---
 src/urma/lib/uvs/core/include/uvs_api.h | 1 +
 src/urma/lib/uvs/core/tpsa_api.c        | 2 +-
 src/urma/tools/urma_admin/admin_cmd.c   | 2 +-
 3 files changed, 3 insertions(+), 2 deletions(-)

diff --git a/src/urma/lib/uvs/core/include/uvs_api.h b/src/urma/lib/uvs/core/include/uvs_api.h
index 8ae1b82..c699516 100644
--- a/src/urma/lib/uvs/core/include/uvs_api.h
+++ b/src/urma/lib/uvs/core/include/uvs_api.h
@@ -78,6 +78,7 @@ int uvs_set_topo_info(void *topo, uint32_t topo_num);
 /**
  * Get primary and port eid from topo info.
  * @param[in] route: parameter that contains src_v_eid and dst_v_eid,
+                            src_v_eid and dst_v_eid can only be eids of bonding devices.
  *                          refers to uvs_route_t;
  * @param[out] route_list: a list buffer, containing all routes returned;
  * Return: 0 on success, other value on error
diff --git a/src/urma/lib/uvs/core/tpsa_api.c b/src/urma/lib/uvs/core/tpsa_api.c
index fec083f..b0aa224 100644
--- a/src/urma/lib/uvs/core/tpsa_api.c
+++ b/src/urma/lib/uvs/core/tpsa_api.c
@@ -58,7 +58,7 @@ int uvs_get_device_name_by_eid(uvs_eid_t *eid, char *buf, size_t len)
 {
     int ret = 0;
 
-    if (buf == NULL || len == 0) {
+    if (buf == NULL || len == 0 || eid == NULL) {
         TPSA_LOG_ERR("Invalid parameter.\n");
         return -EINVAL;
     }
diff --git a/src/urma/tools/urma_admin/admin_cmd.c b/src/urma/tools/urma_admin/admin_cmd.c
index 361e25d..13db742 100644
--- a/src/urma/tools/urma_admin/admin_cmd.c
+++ b/src/urma/tools/urma_admin/admin_cmd.c
@@ -139,7 +139,7 @@ int exec_cmd(admin_config_t *cfg, const admin_cmd_t *cmds)
     }
 
     printf("Unknown cmd '%s'.\n", cmd_name);
-    return 0;
+    return -EINVAL;
 }
 
 bool is_ubc(const char *dev_name)
-- 
2.28.0.windows.1

